flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NettyShuffleEnvironment.java

```diff
@@ -303,7 +309,18 @@ private void registerInputMetrics(MetricGroup inputGroup, MetricGroup buffersGro
 			InputGateMetrics.registerQueueLengthMetrics(inputGroup, inputGates);
 		}
 		buffersGroup.gauge(METRIC_INPUT_QUEUE_LENGTH, new InputBuffersGauge(inputGates));
-		buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new InputBufferPoolUsageGauge(inputGates));
+
+		if (config.isCreditBased()) {
+			FloatingBuffersUsageGauge floatingBuffersUsageGauge = new FloatingBuffersUsageGauge(inputGates);
+			ExclusiveBuffersUsageGauge exclusiveBuffersUsageGauge = new ExclusiveBuffersUsageGauge(inputGates);
+
+			buffersGroup.gauge(METRIC_INPUT_EXCLUSIVE_BUFFERS_USAGE, exclusiveBuffersUsageGauge);
+			buffersGroup.gauge(METRIC_INPUT_FLOATING_BUFFERS_USAGE, floatingBuffersUsageGauge);
+			buffersGroup.gauge(METRIC_INPUT_POOL_USAGE, new CreditBasedInputBufferPoolUsageGauge(floatingBuffersUsageGauge,
+				exclusiveBuffersUsageGauge, inputGates));
```

Comment:
If the parameters are broken into separate lines, make every parameter in a separate line.